### PR TITLE
Update melody accept button for empty string

### PR DESCRIPTION
**Pull request by mathiasvr**
In melody editor when e.g. removing all rtttl (so that the textbox is empty), currentMelody is an empty string and the accept button state is not updated. This allows clicking it which gives an error.

I updated it so it will also consider the empty string, but maybe it would be better to make an else case for `if(currentMelody)`? I'm not sure when/if currentMelody can be `null` or `undefined`?

### CODE REVIEW

**Comment by stylesuxx** [comment]
I am not sure if we should consider an empty melody valid? I'd assume that the firmware can handle an empty melody. In this case it would make sense to consider the empty string as a valid melody to basically allow the user to disable the melody completely

That being said:
```currentMelody``` is set from melody which is a required property. The only point I am not sure is the forth and back conversion through the RTTTL lib. It could be that this at some point returns something else than a string and thus the check is in place.

Long story short, your check is fine IMHO.

**Comment by mathiasvr** [comment]
We need to at least handle the empty string otherwise the accept button stays enabled and throws when pressed.
Might be simpler to remove the if case if we expect it to always be a string but I'm not really sure how the react type stuff works so maybe this is safer.

**Comment by stylesuxx** [comment]
The if can be removed completely. The conversion is wrapped in a try/catch block and if the conversion fails play and accept button will be disabled anyway.

**Comment by mathiasvr** [comment]
Yeah that makes sense, I updated it to remove it. Seems to work fine.
